trestle/core/profile_resolver.py

```diff
@@ -228,90 +304,73 @@ def __init__(self, profile: prof.Profile) -> None:
             logger.debug('merge filter initialize')
             self._profile = profile
 
-        def _join_attr_lists(self, dest: OBT, src: OBT, attr: str) -> None:
-            src_list = getattr(src, attr, None)
-            if not src_list:
-                return
-            dest_list = as_list(getattr(dest, attr, None))
-            dest_list.extend(src_list)
-            setattr(dest, attr, dest_list)
-
-        def _merge_controls(self, dest: cat.Control, src: cat.Control) -> None:
-            """
-            Merge two controls together when the merge method is merge.
-
-            According to OSCAL docs the merge combines all contents of two merged controls
-            even if two items have the same name.  ids, such as control ids, must be unique -
-            but names, such as prop names, may be duplicated.
-            """
-            for attr in ['params', 'props', 'links', 'parts']:
-                self._join_attr_lists(dest, src, attr)
-
-            if src.controls:
-                if not dest.controls:
-                    dest.controls = src.controls
-                else:
-                    self._merge_control_lists(dest.controls, src.controls, prof.Method.merge)
+        def _get_id(self, item: OBT) -> Optional[str]:
+            id_ = getattr(item, ID, None)
+            if id_ is None:
+                id_ = getattr(item, NAME, None)
+            return id_
```

Comment:
Is this something common that can be placed inside OscalBaseModel object? Or is this specific only for the profile?

Reply:
I think this is needed because the effective ID depends on object type.  This is an unusual type of matching by id.